src/V3Param.cpp

```diff
@@ -767,7 +767,12 @@ class ParamProcessor final {
             }
         } else if (AstParamTypeDType* const modvarp = pinp->modPTypep()) {
             AstNodeDType* rawTypep = VN_CAST(pinp->exprp(), NodeDType);
-            AstNodeDType* exprp = rawTypep ? rawTypep->skipRefToNonRefp() : nullptr;
+            AstNodeDType* lastp = nullptr;
+            AstNodeDType* exprp = rawTypep ? rawTypep->skipRefOrNullToNonRefp(&lastp) : nullptr;
+            if (!exprp && lastp) {
+                V3Width::widthParamsEdit(lastp);
+                exprp = lastp->skipRefToNonRefp();
+            }
```

Comment:
Hmm.  Does it work correctly also if we always do V3Width::widthParamsEdit()?

Reply:
Note not suggesting you change it yet, just learning.

Reply:
Yeah, this also works / everything passes:
```
             AstNodeDType* exprp = rawTypep ? rawTypep->skipRefOrNullToNonRefp(&lastp) : nullptr;
-            if (!exprp && lastp) {
+            if (exprp) {
+                V3Width::widthParamsEdit(exprp);
+            } else {
                 V3Width::widthParamsEdit(lastp);
                 exprp = lastp->skipRefToNonRefp();
             }
```

Reply:
Can you just always `widthParamsEdit(exprp)?

My concern otherwise is the lastp might not fully resolve, e.g. might get to another ref and be broken.  widthParamsEdit should be able to handle that itself, and if it does using lastp isn't needed at all.

Reply:
Sorry, I don't follow.  If the type is UNLINKED and skipRef returns `nullptr` what should I do?  Should I always `widthParamsEdit(rawType)`?  Does it work that way?

Reply:
Yes, does this work?

`if (rawType) rawType = widthParamsEdit(rawType);

Reply:
Yes, please.  Don't need lastptr addition then I think.

Reply:
Sorry, I think I'm talking past you since GH isn't showing me these comments until I refresh the page.

Anyway, cool, I'll put this up tomorrow.